internal/vault/client_factory.go

```diff
@@ -270,6 +275,23 @@ func (m *cachingClientFactory) isDisabled() bool {
 	return m.shutDown
 }
 
+func (m *cachingClientFactory) cacheKeyLock(cacheKey ClientCacheKey) (*sync.RWMutex, bool) {
+	m.muLocks.Lock()
+	defer m.muLocks.Unlock()
+	lock, ok := m.cacheKeyLocks[cacheKey]
+	if !ok {
+		lock = &sync.RWMutex{}
+		m.cacheKeyLocks[cacheKey] = lock
+	}
+	return lock, ok
+}
+
+func (m *cachingClientFactory) removeLock(cacheKey ClientCacheKey) {
```

Comment:
This looks pretty specific to the cacheKey/client lock, so maybe something more along the lines of `removeCacheKeyLock()` or `removeClientLock`?

Reply:
Updated in b73710dd31d0aff71e7a57c9ffe4d9d4c321772e